### PR TITLE
add builder for fastp v0.23.3

### PR DESCRIPTION
**Pull request by maltesie**

### CODE REVIEW

File: F/fastp/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+using BinaryBuilder, Pkg
+
+name = "fastp"
+version = v"0.23.3"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/intel/isa-l.git", "2bbce31943289d5696bcf2a433124c50928226a2"),
+    GitSource("https://github.com/ebiggers/libdeflate.git", "02dfa32da3ee3982c66278e714d2e21276dfb67b"),
+    GitSource("https://github.com/OpenGene/fastp.git", "ca559a71feed94e74ea449e7567d0506de48dea4"),
```
Comment (giordano) [inline F/fastp/build_tarballs.jl:12]: Having multiple sources is a good indication you should rather have separate packages.  And we already have [libdeflate](https://github.com/JuliaPackaging/Yggdrasil/blob/dc00ba45dc8953eb72128b72b847963d94bf9405/L/libdeflate/build_tarballs.jl).

File: F/fastp/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+# The products that we will ensure are always built
+products = [
+    ExecutableProduct("fastp", :fastp)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency(PackageSpec(name="NASM_jll", uuid="08ca2550-6d73-57c0-8625-9b24120f3eae"))
+    Dependency(PackageSpec(name="YASM_jll", uuid="997772c2-56d0-5ccd-9329-3f55f14e5768"))
```
Comment (giordano) [inline F/fastp/build_tarballs.jl:54]: Assemblers should typically be [host dependencies](https://docs.binarybuilder.org/stable/build_tips/#Dependencies-for-the-target-system-vs-host-system), not target dependencies

**Review by giordano — CHANGES_REQUESTED** [review]

File: F/fastp/build_tarballs.jl (hunk 1 above)
Comment (maltesie) [reply F/fastp/build_tarballs.jl:12]: I made another pull request for isa-l, will adjust the fastp builder to use the binary dependencies when libisal_jll is available.

File: F/fastp/build_tarballs.jl (hunk 2 above)
Comment (maltesie) [reply F/fastp/build_tarballs.jl:54]: I put them as host dependencies into the libisal builder. Thanks for the comments, I'm learning a lot :) I hope this doesn't take up too much of your time.

**Comment by maltesie** [comment]
Thanks for the help with isa-l, now fastp also compiles for everything, isa-l is available for.

File: F/fastp/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    GitSource("https://github.com/OpenGene/fastp.git", "ca559a71feed94e74ea449e7567d0506de48dea4")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/fastp
+make -j${nproc}
+install -Dvm 755 "fastp${exeext}" "${bindir}/fastp${exeext}"
+mkdir -p "${prefix}/share/licenses/fastp/"
+cp ./LICENSE "${prefix}/share/licenses/fastp/"
```
Comment (giordano) [inline F/fastp/build_tarballs.jl:19]: We have a utility to simplify this: https://docs.binarybuilder.org/stable/build_tips/#Installing-the-license-file
```suggestion
install_license LICENSE
```

File: F/fastp/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    Platform("armv6l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("powerpc64le", "linux"; libc = "glibc"),
+    Platform("x86_64", "linux"; libc = "musl"),
+    Platform("aarch64", "linux"; libc = "musl"),
+    Platform("armv6l", "linux"; call_abi = "eabihf", libc = "musl"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "musl"),
+    Platform("x86_64", "macos"; ),
+    Platform("x86_64", "freebsd"; )
+]
```
Comment (giordano) [inline F/fastp/build_tarballs.jl:36]:
```suggestion
platforms = supported_platforms()
# Same limitations as isa_l
filter!(p -> arch(p) != "i686", platforms)
filter!(!Sys.iswindows, platforms)
filter!(p -> !(Sys.isapple(p) && arch(p) == "aarch64"), platforms)
```

**Review by giordano — APPROVED** [review]

**Pull request merged**